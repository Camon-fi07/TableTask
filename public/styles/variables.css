body {
  --colorTextPrimary: #000000;
  --colorTextPrimaryInvariably: #000000;
  --colorTextSecondary: #818c99;
  --colorTextContrast: #ffffff;
  --colorTextContrastThemed: #ffffff;
  --colorTextAccentThemed: #2688eb;
  --colorBackgroundAccent: #2688eb;
  --colorBackgroundAccentThemed: #2688eb;
  --colorBackgroundAccentTint: #5a9eff;
  --colorBackgroundAccentThemedHover: #2c7cde;
  --colorBackgroundAccentThemedActive: #1e6fdb;
  --colorStrokeAccent: #2688eb;
  --colorStrokeAccentThemed: #2688eb;
  --colorBackgroundNegative: #e64646;
  --colorBackgroundNegativeHover: #d83e3c;
  --colorBackgroundNegativeActive: #c93633;
  --colorBackgroundSecondaryAlpha: #0000000a;
  --colorBackgroundContrastSecondaryAlpha: #ffffff33;
}
