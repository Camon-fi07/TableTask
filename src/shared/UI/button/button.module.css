.wrapper {
  font-weight: 500;
  line-height: 18px;
  font-size: 14px;
  border-radius: 8px;
  margin: 0;
  padding: 0;
  cursor: pointer;
  text-align: center;
}

.small {
  padding: 6px 12px;
}

.large {
  padding: 10px 16px;
}

.wrapper:disabled {
  opacity: 0.64;
}

.primary {
  background-color: var(--colorBackgroundAccentThemed);
  color: var(--colorTextContrastThemed);
}

.primary:hover {
  background-color: var(--colorBackgroundAccentThemedHover);
}

.primary:active {
  background-color: var(--colorBackgroundAccentThemedActive);
}

.negative {
  background-color: var(--colorBackgroundNegative);
  color: var(--colorTextContrast);
}

.negative:hover {
  background-color: var(--colorBackgroundNegativeHover);
}

.negative:active {
  background-color: var(--colorBackgroundNegativeActive);
}

.outline {
  border: 1px solid var(--colorStrokeAccentThemed);
  color: var(--colorTextAccentThemed);
}

.outline:hover {
  background-color: var(--colorBackgroundSecondaryAlpha);
}

.outline:active {
  background-color: var(--colorBackgroundContrastSecondaryAlpha);
}
