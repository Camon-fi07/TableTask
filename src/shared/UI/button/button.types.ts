import { ButtonHTMLAttributes } from 'react';

export type Mode = 'primary' | 'negative' | 'outline';
export type Size = 'small' | 'large';

export interface ButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
  size?: Size;
  mode?: Mode;
}
