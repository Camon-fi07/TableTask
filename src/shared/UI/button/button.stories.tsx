import type { Meta, StoryObj } from '@storybook/react-vite';

import { Button } from './button';

const meta: Meta<typeof Button> = {
  title: 'UI/Button',
  component: Button,
  tags: ['autodocs'],
  argTypes: {
    mode: {
      control: 'select',
      options: ['primary', 'negative', 'outline']
    },
    size: {
      control: 'radio',
      options: ['small', 'large']
    },
    onClick: { action: 'clicked' }
  }
};

export default meta;
type Story = StoryObj<typeof Button>;

export const PrimarySmall: Story = {
  args: {
    children: 'Primary Small',
    mode: 'primary',
    size: 'small'
  }
};

export const PrimaryLarge: Story = {
  args: {
    children: 'Primary Large',
    mode: 'primary',
    size: 'large'
  }
};

export const NegativeSmall: Story = {
  args: {
    children: 'Negative Small',
    mode: 'negative',
    size: 'small'
  }
};

export const OutlineLarge: Story = {
  args: {
    children: 'Outline Large',
    mode: 'outline',
    size: 'large'
  }
};
