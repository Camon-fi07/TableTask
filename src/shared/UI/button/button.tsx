import styles from './button.module.css';
import { ButtonProps } from './button.types';
import { getModeClassName, getSizeClassName } from './button.utils';

export const Button = ({
  className,
  size = 'small',
  mode = 'primary',
  ...props
}: ButtonProps) => {
  return (
    <button
      className={`${className || ''} ${getSizeClassName(size)} ${getModeClassName(mode)} ${styles.wrapper}`}
      {...props}
    />
  );
};
