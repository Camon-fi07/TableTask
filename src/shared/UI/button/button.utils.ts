import styles from './button.module.css';
import { Mode, Size } from './button.types';

export const getSizeClassName = (size: Size) => {
  switch (size) {
    case 'small':
      return styles.small;
    case 'large':
      return styles.large;
    default:
      return styles.small;
  }
};

export const getModeClassName = (mode: Mode) => {
  switch (mode) {
    case 'primary':
      return styles.primary;
    case 'outline':
      return styles.outline;
    case 'negative':
      return styles.negative;
    default:
      return styles.primary;
  }
};
