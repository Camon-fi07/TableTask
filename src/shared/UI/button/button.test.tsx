import { render, screen } from '@testing-library/react';

import { Button } from './button';
import styles from './button.module.css';

describe('Button', () => {
  it('should render', () => {
    render(<Button>Click me</Button>);
    expect(screen.getByRole('button')).toBeInTheDocument();
  });

  it('should match snapshot', () => {
    render(<Button>Click me</Button>);
    expect(screen.getByRole('button')).toMatchSnapshot();
  });

  it('should apply size and mode', () => {
    render(
      <Button size='large' mode='negative'>
        Test
      </Button>
    );

    const btn = screen.getByRole('button');

    expect(btn).toHaveClass(styles.large);
    expect(btn).toHaveClass(styles.negative);
  });
});
