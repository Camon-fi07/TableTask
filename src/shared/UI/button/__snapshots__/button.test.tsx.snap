// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should match snapshot 1`] = `
<button
  class=" small primary wrapper"
>
  Click me
</button>
`;
